src/DateInterval.php

```diff
@@ -12,15 +12,19 @@ class DateInterval extends BaseDateInterval
     /**
      * Create a DateInterval object from string and throw exception if invalid interval provided
      *
-     * @param string $interval Interval spec string
+     * @param \DateInterval|string $interval Interval spec string
      *
      * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeIntervalException
      */
-    public function __construct(string $interval)
+    public function __construct($interval)
     {
         try {
             // Create parent object
-            parent::__construct($interval);
+            parent::__construct(
```

Comment:
This could result in a fatal if you pass in another object type.

I think you should check the instanceof before the parent call and override interval with the ->format, and if it isnt a string at that point throw an InvalidDateTimeIntervalException

Reply:
This applies to all constructor changes. We're going to need to check and test for any input values now that it accepts more than just string|null, including ints, other objects, etc.